src/snakeoil/dist/distutils_extensions.py

```diff
@@ -623,12 +623,12 @@ class build_scripts(dst_build_scripts.build_scripts):
 
     def finalize_options(self):
         super().finalize_options()
-        script_dir = os.path.join(
+        self.script_dir = os.path.join(
             os.path.dirname(self.build_dir), '.generated_scripts')
-        self.mkpath(script_dir)
-        self.scripts = [os.path.join(script_dir, x) for x in os.listdir(SCRIPTS_DIR)]
+        self.scripts = [os.path.join(self.script_dir, os.path.basename(x)) for x in self.scripts]
```

Comment:
Well, it's of no consequence right now but maybe we should be using `relpath()` to `SCRIPTS_DIR` rather than `basename()`?

Reply:
There's no requirement in `build_scripts` that script content come from `SCRIPTS_PATH`; that's just a hack that was shoved into this subcommand (breaking other things in the process, as I showed).

What I'd suggest doing is renaming this subcommand to `build_script_redirects` rather than breaking the abilty to use `build_scripts` as it was intended, but much like my other comment, my suggestion here is "restore functionality and usability, then refactor/cleanup".